opentelemetry/src/global/metrics.rs

```diff
@@ -19,10 +19,15 @@ pub fn set_meter_provider<P>(new_provider: P)
 where
     P: metrics::MeterProvider + Send + Sync + 'static,
 {
-    let mut global_provider = global_meter_provider()
-        .write()
-        .expect("GLOBAL_METER_PROVIDER RwLock poisoned");
-    *global_provider = Arc::new(new_provider);
+    // Try to set the global meter provider. If the RwLock is poisoned, we'll log an error.
+    let mut global_provider = global_meter_provider().write();
+
+    if let Ok(ref mut provider) = global_provider {
+        **provider = Arc::new(new_provider);
+        otel_info!(name: "MeterProvider.GlobalSet", message = "Global meter provider is set. Meters can now be created using global::meter() or global::meter_with_scope().");
+    } else {
+        otel_error!(name: "MeterProvider.GlobalSetFailed", message = "Global meter provider is not set due to lock poison. Meters created using global::meter() or global::meter_with_scope() will not function.");
```

Comment:
not sure if this (lock poison) is ever user-actionable... Maybe just state "OTel internal error occurred, report it at OpenTelemtry Github repo"?

Reply:
It's not user-actionable in its current form. If `set_meter_provider` returned a `PoisonError`, users could reinitialize the provider to Noop and retry. In current form, it's better to log the error and instruct users to report it to the repository.

Reply:
will change it in next PR